Added es|ql queries to tsdb k8s query track.

martijnvg opened the pull request:
This PR adds es|ql queries to the tsdb k8s track based on the following already defined searches: `esql_cpu_usage_per_pod_*`, `memory_usage_per_pod_*`, `average_container_cpu_*`, `average_container_memory_usage_*`, `cpu_usage_per_container_*`, `memory_usage_per_container_*`, `unique_deployment_count_*` and `percentile_cpu_usage_per_container_*`.

There is one caveat with the newly defined es|ql queries and that is that es|ql queries group by pod/container and then time interval. Whereas the searches also group by just pod/container. For example `cpu_usage_per_pod_15_minutes` computes average cpu usage by pod and 30 second interval and by pod, whereas the esql variant only computes cpu usage by pod and 30 second interval. In es|ql grouping currently can only be done by one set of aggregation keys. Maybe supporting multiple sets of aggregation kets will be supported in the future by the inline stats command.

Additionally the following searches couldn't be ported to esql queries:

- The `status_per_pod_*` searches, because there is no equivalent yet for the `top_metrics` aggregation in es|ql.
- The `tx_network_usage_per_pod_*` searches, because this search uses an average aggregation on a counter field and es|ql doesn't allow stats operations to be performed yet on counter fields.

gareth-ellis reviewed (commented):
Hello,

I have a couple of comments:
- By adding es|ql after each of the existing queries, the track will no longer be strictly comparable with previous runs - have you considered adding the es|ql portion at the end of the track or even as a new challenge? 
- Have you run this already somewhere, and do you know the runtime? The track currently takes around 3 hours to run, so I would guess it may be close to double that after this change, given most queries are run both as previously and then as es|ql.

martijnvg commented:
Thanks for taking a look @gareth-ellis. I just checked the rally log when I was running this track with new changes in esbench. And now it takes roughly 5 hours to complete this track. So I think the benchmark time is getting out of hand. I like the idea of bundling the esql queries in a new track. I will update the PR to do this.

martijnvg commented:
I've updated the PR and moved the esql queries into a new challenge (instead of a new track). This way a number of things get reused, which it think makes sense (esql queries use the same corpora.

gareth-ellis reviewed (approved):
LGTM

martijnvg commented:
thanks @gareth-ellis!
